Tasks/VsTest/vstest.ts

```diff
@@ -65,17 +67,20 @@ export function startTest() {
                     tl.setResult(code, tl.loc('VstestReturnCode', code));
                     deleteVstestDiagFile();
                 } catch (error) {
+                    publishEventToCi(areaCodes.publishResults, error.message);
                     deleteVstestDiagFile();
                     console.log('##vso[task.logissue type=error;code=' + error + ';TaskName=VSTest]');
-                    throw error;
+                    throw error; // This throw should be removed
                 }
             })
             .fail(function (err) {
+                publishEventToCi(areaCodes.invokeVsTest, err.message);
                 deleteVstestDiagFile();
                 console.log('##vso[task.logissue type=error;code=' + err + ';TaskName=VSTest]');
-                throw err;
+                throw err;  // What is the point of this throw? Where is it getting caught?
             });
     } catch (error) {
+        publishEventToCi(areaCodes.runTestsLocally, error.message);
```

Comment:
JSON.stringfy

Reply:
done

---
In reply to: [136252940](https://github.com/Microsoft/vsts-tasks/pull/5156#discussion_r136252940) [](ancestors = 136252940)

Reply:
This doesn't get the message out. Reverting to error.message

---
In reply to: [136255079](https://github.com/Microsoft/vsts-tasks/pull/5156#discussion_r136255079) [](ancestors = 136255079,136252940)